[BUG]: `sonatyperepo_capability_webhook_*` resources fail against NXRM 3.87.1

### Terraform Version

1.13.5

### Provider Version

0.17.0

### Sonatype Nexus Repository Manager version

3.87.1

### High Availability enabled?

No - Single node only (Default)

### Describe the Issue or Bug

Acc Tests fail against NXRM 3.87.1:
- TestAccCapabilityWebhookGlobalResource
- TestAccCapabilityWebhookRepositoryResource

### Terraform Configuration Files

```hcl
resource "sonatyperepo_capability_webhook_global" "webhook" {
  enabled = true
  notes = "These are notes from Terraform"
  properties = {
    names = [ "repository" ]
    url = "https://test.tld"
    secret = "testing"
  }
}
```

### Steps to Reproduce

Run acc tests.

### Logs & Output

```shell
--- FAIL: TestAccCapabilityWebhookRepositoryResource (0.74s)
    capability_x_resource_test.go:504: Step 1/1 error: Error running apply: exit status 1
        
        Error: Provider produced inconsistent result after apply
        
        When applying changes to sonatyperepo_capability_webhook_repository.cap,
        provider "provider[\"registry.terraform.io/hashicorp/sonatyperepo\"]"
        produced an unexpected new value: .properties: inconsistent values for
        sensitive attribute.
        
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
FAIL
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/internal/provider/capability/capability_common.go
+++ b/internal/provider/capability/capability_common.go
@@ -103,8 +103,9 @@ func (c *capabilityResource) Create(ctx context.Context, req resource.CreateRequ
 		)
 	}
 
-	plan = c.CapabilityType.UpdateStateFromApi(plan, capabilityCreateResponse)
-	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+	stateModel := c.CapabilityType.UpdateStateFromApi(plan, capabilityCreateResponse)
+	stateModel = c.CapabilityType.MapFromPlanToState(plan, stateModel)
+	resp.Diagnostics.Append(resp.State.Set(ctx, stateModel)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
@@ -163,8 +164,9 @@ func (c *capabilityResource) Read(ctx context.Context, req resource.ReadRequest,
 		return
 	}
 
-	stateModel = c.CapabilityType.UpdateStateFromApi(stateModel, capability)
-	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+	currentStateModel := c.CapabilityType.UpdateStateFromApi(stateModel, capability)
+	currentStateModel = c.CapabilityType.MapFromPlanToState(stateModel, currentStateModel)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &currentStateModel)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
@@ -246,6 +248,7 @@ func (c *capabilityResource) Update(ctx context.Context, req resource.UpdateRequ
 	}
 
 	stateModel = c.CapabilityType.UpdateStateFromApi(stateModel, capability)
+	stateModel = c.CapabilityType.MapFromPlanToState(planModel, stateModel)
 	resp.Diagnostics.Append(resp.State.Set(ctx, stateModel)...)
 	if resp.Diagnostics.HasError() {
 		return

--- a/internal/provider/capability/capability_type/common.go
+++ b/internal/provider/capability/capability_type/common.go
@@ -66,6 +66,10 @@ func (ct *BaseCapabilityType) GetType() common.CapabilityType {
 	return ct.capabilityType
 }
 
+func (ct *BaseCapabilityType) MapFromPlanToState(plan any, state any) any {
+	return state
+}
+
 // --------------------------------------------
 // CapabilityTypeI that all Capability Types must implement
 // --------------------------------------------
@@ -83,6 +87,7 @@ type CapabilityTypeI interface {
 	GetType() common.CapabilityType
 	UpdatePlanForState(plan any) any
 	UpdateStateFromApi(state any, api any) any
+	MapFromPlanToState(plan any, state any) any
 	UpdateStateFromPlanForUpdate(plan any, state any) any
 }
 

--- a/internal/provider/capability/capability_type/webhook_global.go
+++ b/internal/provider/capability/capability_type/webhook_global.go
@@ -93,6 +93,13 @@ func (f *WebhookGlobalCapability) UpdateStateFromApi(state any, api any) any {
 	return stateModel
 }
 
+func (ct *WebhookGlobalCapability) MapFromPlanToState(plan any, state any) any {
+	planModel := (plan).(model.WebhookGlobalCapabilityModel)
+	stateModel := (state).(model.WebhookGlobalCapabilityModel)
+	stateModel.Properties.Secret = types.StringValue(planModel.Properties.Secret.ValueString())
+	return stateModel
+}
+
 func (f *WebhookGlobalCapability) UpdateStateFromPlanForUpdate(plan any, state any) any {
 	planModel := (plan).(model.WebhookGlobalCapabilityModel)
 	stateModel := (state).(model.WebhookGlobalCapabilityModel)

--- a/internal/provider/capability/capability_type/webhook_repository.go
+++ b/internal/provider/capability/capability_type/webhook_repository.go
@@ -93,6 +93,13 @@ func (f *WebhookRepositoryCapability) UpdateStateFromApi(state any, api any) any
 	return stateModel
 }
 
+func (ct *WebhookRepositoryCapability) MapFromPlanToState(plan any, state any) any {
+	planModel := (plan).(model.WebhookRepositoryCapabilityModel)
+	stateModel := (state).(model.WebhookRepositoryCapabilityModel)
+	stateModel.Properties.Secret = types.StringValue(planModel.Properties.Secret.ValueString())
+	return stateModel
+}
+
 func (f *WebhookRepositoryCapability) UpdateStateFromPlanForUpdate(plan any, state any) any {
 	planModel := (plan).(model.WebhookRepositoryCapabilityModel)
 	stateModel := (state).(model.WebhookRepositoryCapabilityModel)

--- a/internal/provider/model/capability_webhook.go
+++ b/internal/provider/model/capability_webhook.go
@@ -81,11 +81,7 @@ func (m *WebhookGlobalCapabilityModel) FromApiModel(api *v3.CapabilityDTO) {
 		namesValues = append(namesValues, types.StringValue(strings.TrimSpace(part)))
 	}
 	m.Properties.Names = types.SetValueMust(types.StringType, namesValues)
-	if secret, ok := (*api.Properties)["secret"]; ok && secret != "" {
-		m.Properties.Secret = types.StringValue(secret)
-	} else {
-		m.Properties.Secret = types.StringNull()
-	}
+	// Skip Password
 	m.Properties.Url = types.StringValue((*api.Properties)["url"])
 }
 
@@ -161,11 +157,7 @@ func (m *WebhookRepositoryCapabilityModel) FromApiModel(api *v3.CapabilityDTO) {
 		namesValues = append(namesValues, types.StringValue(strings.TrimSpace(part)))
 	}
 	m.Properties.Names = types.SetValueMust(types.StringType, namesValues)
-	if secret, ok := (*api.Properties)["secret"]; ok && secret != "" {
-		m.Properties.Secret = types.StringValue(secret)
-	} else {
-		m.Properties.Secret = types.StringNull()
-	}
+	// Skip Password
 	m.Properties.Url = types.StringValue((*api.Properties)["url"])
 	m.Properties.Repository = types.StringValue((*api.Properties)["repository"])
 }

